File: Anul II/Medii de proiectare si programare/Laborator/lab07/Ubb.BikeContest.Server/ContestConcurrentServer.cs
using System.Net.Sockets;
using Ubb.BikeContest.Networking;

namespace Ubb.BikeContest.Services
{
    public class ContestConcurrentServer : ConcurrentServer
    {
        private readonly IContestServices server;
        private ClientWorker? worker;

        public ContestConcurrentServer(string host, int port, IContestServices server) : base(host, port)
        {
            this.server = server;
            Console.WriteLine("Server started...");
        }

        protected override Thread CreateWorker(TcpClient client)
        {
            worker = new ClientWorker(server, client);
            return new Thread(new ThreadStart(worker.Run));
        }
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab09/server/src/main/java/com/ubb/server/ContestConcurrentServer.java
package com.ubb.server;

import com.ubb.IContestServices;
import com.ubb.protocol.ClientWorker;

import java.net.Socket;

public class ContestConcurrentServer extends ConcurrentServer {
    private final IContestServices contestServer;

    public ContestConcurrentServer(int port, IContestServices contestServer) {
        super(port);
        this.contestServer = contestServer;
    }

    @Override
    protected Thread createWorker(Socket client) {
        ClientWorker worker = new ClientWorker(contestServer, client);
        return new Thread(worker);
    }
}
